File: src/types/reservationMealPeriod.ts
export type ReservationMealPeriodStatus = 'active' | 'disabled'

export interface ReservationMealPeriod {
  id: string
  periodKey: string
  displayName: string
  startLocalTime: string
  endLocalTime: string
  crossesNextDay: boolean
  slotIntervalMinutes: number
  status: ReservationMealPeriodStatus
  sortOrder: number
  version: number
}

export interface ReservationMealPeriodMutation {
  periodKey: string
  displayName: string
  startLocalTime: string
  endLocalTime: string
  crossesNextDay: boolean
  slotIntervalMinutes: number
  status: ReservationMealPeriodStatus
  sortOrder: number
  version?: number
}

export interface PlatformReservationMealPeriodSeedMutation {
  periods: ReservationMealPeriodMutation[]
}

export interface PlatformReservationMealPeriodSeedResponse {
  success: true
  periods: ReservationMealPeriod[]
}

export interface ReservationMealPeriodSettingsMutation {
  usePlatformSeed: boolean
  copyPlatformSeed?: boolean
  periods?: ReservationMealPeriodMutation[]
}

export interface ReservationMealPeriodSettingsResponse {
  success: true
  usePlatformSeed: boolean
  platformPeriods: ReservationMealPeriod[]
  storePeriods: ReservationMealPeriod[]
  effectivePeriods: ReservationMealPeriod[]
}

export interface ReservationTimeSlot {
  periodId: string
  periodKey: string
  displayName: string
  businessDate: string
  time: string
  startAt: string
  nextDay: boolean
  selectable: boolean
}

export interface ReservationTimeSlotListResponse {
  success: true
  storeId: string
  businessDate: string
  timezone: string
  slots: ReservationTimeSlot[]
}

export interface ReservationMealPeriodApiErrorResponse {
  success: false
  error: {
    code: string
    messageKey: string
    details: Record<string, unknown>
  }
}

File: src/main/java/com/rpb/reservation/reservation/application/ReservationMealPeriod.java
package com.rpb.reservation.reservation.application;

import java.time.LocalTime;
import java.util.UUID;

public record ReservationMealPeriod(
    UUID id,
    String periodKey,
    String displayName,
    LocalTime startLocalTime,
    LocalTime endLocalTime,
    boolean crossesNextDay,
    int slotIntervalMinutes,
    String status,
    int sortOrder,
    int version
) {
    public ReservationMealPeriod {
        if (id == null) {
            throw new IllegalArgumentException("meal_period_id_required");
        }
        if (periodKey == null || periodKey.isBlank()) {
            throw new IllegalArgumentException("meal_period_key_required");
        }
        if (displayName == null || displayName.isBlank()) {
            throw new IllegalArgumentException("meal_period_display_name_required");
        }
        if (startLocalTime == null || endLocalTime == null) {
            throw new IllegalArgumentException("meal_period_time_required");
        }
        if (slotIntervalMinutes < 5 || slotIntervalMinutes > 240) {
            throw new IllegalArgumentException("meal_period_interval_invalid");
        }
        if (status == null || status.isBlank()) {
            throw new IllegalArgumentException("meal_period_status_required");
        }
        periodKey = periodKey.trim();
        displayName = displayName.trim();
        status = status.trim();
    }

    public boolean active() {
        return "active".equals(status);
    }
}
